lib/rubocop/cop/lint/out_of_range_regexp_ref.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cops looks for out of range referencing for Regexp, as while capturing groups out of
```

Comment:
I'm not a native speaker, but the "as while" sounds strange to me. How about?

"This cops looks for references of Regexp captures that are out of range and thus always returns nil."